### PR TITLE
add xpu cumprod, group norm grad

### PR DESCRIPTION
**Pull request by ykkk2333**
### PR types

Others 

### PR changes

OPs 

### Describe

add xpu cumprod kernel
add xpu group norm grad kernel
delete pool_grad extra flag

### CODE REVIEW

File: paddle/phi/backends/xpu/xpu2_op_list.cc [hunk 1]
```diff
@@ -712,6 +716,14 @@ XPUOpMap& get_kl2_ops() {
       {"tanh", XPUKernelSet({phi::DataType::FLOAT32, phi::DataType::FLOAT16})},
       {"temporal_shift", XPUKernelSet({phi::DataType::FLOAT32})},
       {"temporal_shift_grad", XPUKernelSet({phi::DataType::FLOAT32})},
+      {"transfer_dtype",
+       XPUKernelSet({phi::DataType::FLOAT32,
+                     phi::DataType::FLOAT16,
+                     phi::DataType::FLOAT64,
+                     phi::DataType::BOOL,
```
Comment (ZibinGuo) [inline paddle/phi/backends/xpu/xpu2_op_list.cc:725]: 加INT8类型 参考cast

File: paddle/phi/kernels/xpu/cumprod_kernel.cc [hunk 2]
```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
```
Comment (ZibinGuo) [inline paddle/phi/kernels/xpu/cumprod_kernel.cc]: 2023

File: paddle/phi/kernels/xpu/cumprod_kernel.cc [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  using XPUType = typename XPUTypeTrait<T>::Type;
+  const DenseTensor* x = &input;
+  auto* x_data = x->data<T>();
+  auto* out_data = dev_ctx.template Alloc<T>(out);
+  DDim shape = x->dims();
+  std::vector<int64_t> xshape = phi::vectorize<int64_t>(shape);
+
+  if (dim < 0) dim += xshape.size();
+  if (shape.size() == 0) {
+    phi::Copy<Context>(dev_ctx, input, dev_ctx.GetPlace(), false, out);
```
Comment (ZibinGuo) [inline paddle/phi/kernels/xpu/cumprod_kernel.cc]: 之前说phi::copy xpu2xpu有问题 需要改用xpu::copy

**Review by ZibinGuo — APPROVED** [review]
LGTM

File: paddle/phi/kernels/xpu/pool_grad_kernel.cc [hunk 4]
```diff
@@ -116,8 +116,7 @@ void Pool2dGradKernel(const Context& ctx,
       // and broadcast kernels to get same output, but better performance.
       // Since the dim is special in particular models,
       // use 'export XPU_POOLING_GRAD_SPECIAL=1' to open this path
```
Comment (XiaociZhang) [inline paddle/phi/kernels/xpu/pool_grad_kernel.cc:118]: 注释一起删掉

**Review by QingshuChen — APPROVED** [review]

**Pull request merged**